include/exadg/incompressible_navier_stokes/time_integration/time_int_bdf.h

```diff
@@ -132,6 +140,39 @@ class TimeIntBDF : public TimeIntBDFBase<Number>
   std::shared_ptr<HelpersALE<Number> const> helpers_ale;
 
 private:
+  template<typename Lambda>
+  void
+  get_quantity_and_times(std::vector<VectorType const *> & quantities,
+                         std::vector<double> &             times,
+                         Lambda const &                    get_quantity) const
+  {
+    unsigned int const current_order = this->get_current_order();
+
+    quantities.resize(current_order);
+    this->fill_at_previous_times(quantities.begin(), quantities.end(), get_quantity);
```

Comment:
I do not really see a value in calling this function over running a for-loop explicitly. Having a template `Lambda` here makes the code unnecessarily complex to understand in my opinion.

Reply:
[1f39cef](https://github.com/exadg/exadg/pull/458/commits/1f39cef8d688232b73b239620a45f409a169e862)